test/common/upstream/load_balancer_impl_test.cc

```diff
@@ -1532,6 +1533,30 @@ TEST_P(LeastRequestLoadBalancerTest, WeightImbalance) {
   EXPECT_EQ(hostSet().healthy_hosts_[0], lb_.chooseHost(nullptr));
 }
 
+TEST_P(LeastRequestLoadBalancerTest, WeightImbalanceWithCustomExponent) {
+  EXPECT_CALL(runtime_.snapshot_,
+              getDouble("upstream.least_requests.active_requests_exponent", 1.0))
+      .WillRepeatedly(Return(0.0));
+
+  hostSet().healthy_hosts_ = {makeTestHost(info_, "tcp://127.0.0.1:80", 1),
+                              makeTestHost(info_, "tcp://127.0.0.1:81", 2)};
+  stats_.max_host_weight_.set(2UL);
+
+  hostSet().hosts_ = hostSet().healthy_hosts_;
+  hostSet().runCallbacks({}, {}); // Trigger callbacks. The added/removed lists are not relevant.
+
+  EXPECT_CALL(random_, random()).WillRepeatedly(Return(0));
+
+  // We should see 2:1 ratio for hosts[1] to hosts[0], regardless of the active request count.
+  hostSet().healthy_hosts_[1]->stats().rq_active_.set(1);
```

Comment:
I'm surprised this works. Are the stats affecting the outstanding requests perceived by the LB when the exponent is non-zero?

Reply:
Yes, the active rq stat is used to calculate the effective weights when the exponent is > 0: https://github.com/envoyproxy/envoy/pull/11252/files/0a2b1fcdad5daeef1136ef7c2208f356d3fcf05f#diff-e746c5f817e7b535220c275b9dea666bR494